## Linear Mixed Models (LMMs)

* LIMIX
  * [preprint](https://www.biorxiv.org/content/10.1101/003905v2),
  * [wiki (LIMIX for eQTL mapping)](https://github.com/single-cell-genetics/limix_qtl/wiki).
* BOLT-LMM
  * [paper](https://www.nature.com/articles/ng.3190) (Loh et al, Nature Genetics 2015),
  * [paper 2](https://www.nature.com/articles/s41588-018-0144-6) (Loh et al, Nature Genetics 2018),
  * [manual](https://alkesgroup.broadinstitute.org/BOLT-LMM/BOLT-LMM_manual.html).
* LEMMA (Linear Environment Mixed Model Analysis)
  * [paper](https://www.sciencedirect.com/science/article/pii/S0002929720302779) (Kerin & Marchinin, AJHG 2020),
  * [code](https://mkerin.github.io/LEMMA/).
* fastGWA
  * [paper](https://www.nature.com/articles/s41588-019-0530-8)
  * [software](https://yanglab.westlake.edu.cn/software/gcta/#fastGWA)
* SAIGE (Scalable and Accurate Implementation of GEneralized mixed model)
  * [paper](https://www.nature.com/articles/s41588-018-0184-y)
  * [code](https://github.com/weizhouUMICH/SAIGE/)
* regenie
  * [paper](https://www.nature.com/articles/s41588-021-00870-7)
  * [webpage](https://rgcgithub.github.io/regenie/)
* GridLMM
  * [code](https://github.com/deruncie/GridLMM)
* [Estimation and Inference for Very Large Linear Mixed Effects Models](https://arxiv.org/pdf/1610.08088.pdf)
* [lme4](https://github.com/lme4/lme4)
* [mgcv](https://www.maths.ed.ac.uk/~swood34/mgcv/#:~:text=mgcv%20is%20an%20R%20package,splines%20with%20automatic%20smoothness%20estimation.)


## Rare Variants (RV) association methods

* SKAT / SKAT-O / burden tests
* LRTq (specific for expression association, i.e., identification of RV-eGenes)
  * [code](https://github.com/avallonking/LRTq),
  * [paper](https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1009596) (Li et al, Plos Genetics 2021).

Adding to [this doc](https://docs.google.com/document/d/1zWrtY_1xe0Ye63ukfwb940g1mLwsvRDe-LgHkDyOzvs/edit) for an overview of methods to detect the effects of RV on expression specifically.

## eQTL methods

* Matrix eQTL
  * [paper](https://academic.oup.com/bioinformatics/article/28/10/1353/213326) (Shabalin, Bioinformatics 2012),
  * [webpage](http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/).
* Tensor QTL
  * [code](https://github.com/broadinstitute/tensorqtl),
  * [paper](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1836-7) (Taylor-Weiner*, Aguet* et al., Genome Biology 2019).
* QTLtools
  * [webpage](https://qtltools.github.io/qtltools/),
  * [paper](https://www.nature.com/articles/ncomms15452) (Delaneau et al., Nature Communications 2017).
* [HEFT](https://academic.oup.com/bioinformatics/article/30/3/369/228688) - webpage no longer exists?
* LORSEN
  * [paper](https://www.frontiersin.org/articles/10.3389/fgene.2021.690926/full)
  * [repo](https://github.com/gaochengPRC/LORSEN)
